File: Backend/Dtos/NewEditRecipeDto.cs

using System.ComponentModel.DataAnnotations;

namespace Backend.Dtos;

public class NewEditRecipeDto
{
    [StringLength(30, MinimumLength = 2)]
    public required string Title { get; set; }

    public required IEnumerable<NewEditRecipeIngredientDto> RecipeIngredients { get; set; }

    public required IEnumerable<NewEditUnitsRecipeDto> UnitsRecipe { get; set; }
}

File: Frontend/src/domain/models/newEditRecipeDto.ts
import type { NewEditRecipeIngredientDto } from "./newEditRecipeIngredientDto";
import type { NewEditUnitsRecipeDto } from "./newEditUnitsRecipeDto";

export interface NewEditRecipeDto {
  title: string;
  recipeIngredients: Array<NewEditRecipeIngredientDto>;
  unitsRecipe: Array<NewEditUnitsRecipeDto>;
}
